### DISCUSSION TITLE
Response from transportOrders/{NAME} web service is empty

### DISCUSSION THREAD

**Discussion by jfaigan** [openTCS/opentcs#50] [Q&A] [answered]
According to the Swagger for the REST API, the transportOrders/{NAME} method should return the details of the created transportOrder when the Status is 200
![image](https://user-images.githubusercontent.com/14929031/194332215-8466d93c-7f40-4f09-8b00-c8b968d6fe1c.png)

However, I am not receiving any info when I try to call this method from Postman.  The status is 200 and the transportOrder is properly created and succeeds...there is just no feedback via the API.

Is this normal?

Regards,
Joel

**Comment by swltr** [comment, marked as answer]
@jfaigan No, this is not normal. I have just tried this (with RESTClient in Firefox), and I am unable to reproduce the behaviour you described. I do get a JSON body containing a description of the created transport order.

**Reply by jfaigan** [reply]
Hi Stefan,

Ok, that is what I figured, however, I am at a loss as to why I would not receive feedback via Postman.  I will try again with another method to see if I can replicate the behavior.  I do not know RESTClient with Firefox but I will take a look.

Thank you for the feedback.  I will post back here with my findings.

Joel

**Reply by jfaigan** [reply]
Hi Stefan,

No.  It does not work for me.

I tried again with Postman:
![image](https://user-images.githubusercontent.com/14929031/196981679-66919890-6678-4b36-aa77-00497403c5e6.png)
![image](https://user-images.githubusercontent.com/14929031/196980825-2bf8dbfd-40d1-44b3-baef-61227e85b4e1.png)

I tried with an Edge plug-in:
![image](https://user-images.githubusercontent.com/14929031/196981009-deabed88-ba78-498d-a7cc-e886488e9b78.png)
![image](https://user-images.githubusercontent.com/14929031/196981060-b2faeaca-6660-43f1-ae2a-7639e5ba128a.png)

And I tried with the Firefox plug-in:
![image](https://user-images.githubusercontent.com/14929031/196981169-31892ebc-ddf5-4858-a492-6afb0298fd3d.png)
![image](https://user-images.githubusercontent.com/14929031/196981214-0f79cfce-b2a3-4ccf-ae41-b4f34431d32c.png)


In all cases, I have a Status 200 and the Transport Order is received and executed by the Operations Desk software, however, there is no feedback via the API.

Is there something you can do to help with this?

Regards,
Joel

**Reply by swltr** [reply]
What's interesting is that it returns `text/plain` as the content type. It should be `application/json`.

* Which version of openTCS are you using?
* Have you made any modifications, especially to the web API code?
* Are there any suspicious messages in the kernel's log file that could be related to this?

**Reply by jfaigan** [reply]
Hi Stefan,

I am using the following version:
![image](https://user-images.githubusercontent.com/14929031/197005336-c5968e9b-c195-4aba-9a06-7dbd06982ebd.png)

I did not change any code...I am a former .Net dev and not a Java dev.  😄 

Everything in the Kernel seems fine.
![image](https://user-images.githubusercontent.com/14929031/197005630-65c36929-cf84-461f-9a74-e4f0616f2e2e.png)

I had accidentally sent the same TO name in my second test and the Kernel correctly showed that error.  You can see in the previous screenshot the successful execution of the tests 2 and 3.
![image](https://user-images.githubusercontent.com/14929031/197005896-a2e14d34-d347-4ee2-9d63-da69615ec9ae.png)

**Reply by swltr** [reply]
Ah. 5.2 did not return a description of the created order, yet. That was introduced with 16973199e62a2f6a8942be7913af6bb398ce2ff4. I guess the documentation you looked was not the one for 5.2. Updating to a more recent version should fix this.

**Reply by jfaigan** [reply]
Ok, that makes sense.  I knew there would be a logical explanation.

Let me see what the upgrade path looks like and then test.

If it works, I will come back and mark this as solved.

Thanks for the assistance.

Joel

**Reply by jfaigan** [reply]
Ok, I can confirm that with the latest version, I am receiving the expected behaviour.

Thank you.
